staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -110,23 +115,11 @@ type Owner struct {
 // without an apiserver. In that case we can't and don't need to publish
 // ResourceSlices.
 func StartController(ctx context.Context, kubeClient kubernetes.Interface, driver string, owner Owner, resources *DriverResources) *Controller {
-	if kubeClient == nil {
-		return nil
-	}
-
 	logger := klog.FromContext(ctx)
-	ctx, cancel := context.WithCancelCause(ctx)
-
-	c := &Controller{
-		cancel:     cancel,
-		kubeClient: kubeClient,
-		driver:     driver,
-		owner:      owner,
-		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
-			workqueue.DefaultTypedControllerRateLimiter[string](),
-			workqueue.TypedRateLimitingQueueConfig[string]{Name: "node_resource_slices"},
-		),
-		resources: resources,
+	c, err := newController(ctx, kubeClient, driver, owner, resources)
+	if err != nil {
+		logger.Error(err, "Failed to create controller")
```

Comment:
As it can fail now, `StartController` should have an additional `error` return value. Returning nil without an error is potentially going to cause panics in the caller.